### PR TITLE
feat: Minting an NFT example

### PR DESCRIPTION
**Pull request by gbibeaul**
# Minting NFTs with Next.js

This example shows how to build an NFT minting platform using Next.js This example contains the following:

- NFT collections infinite scroll to explore different Arts
- Connecting and disconnecting a Metamask Wallet
- Client side Image resizing and validation to keep things family friendly.
- Soft Minting NFTs using Rarible's and Moralis' SDKs

## Deployed Live on Vercel

https://nft-mint-guivercel-vercel-solutions-vtest314.vercel.app/

## Getting Started

1. Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:

```bash
npx create-next-app --example https://github.com/vercel/examples/tree/main/solutions/nft-mint nft-mint
# or
yarn create next-app --example https://github.com/vercel/examples/tree/main/solutions/nft-mint nft-mint
```

2. Register and deploy a [Moralis Server](https://admin.moralis.io/register).
3. Install the [Rarible Plugin on your Moralis server](https://moralis.io/plugins/rarible-nft-tools/). You will mneed the information about [Moralis Speedy Nodes](https://docs.moralis.io/speedy-nodes/connecting-to-rpc-nodes/connect-to-eth-node) to deploy the plugin to your Server. Make sure you pick the Ethereum Network of your choice.
4. Fill your environment variables with the Moralis server values. These values can safely be public:

```
NEXT_PUBLIC_SERVER_URL=<server url goes here>
NEXT_PUBLIC_APP_ID=<Moralis app id goes here>
```

## Built with Open source projects:

- [NFSW JS](https://nsfwjs.com/) for Image validation
- [Tailwindcss](https://tailwindcss.com/) for styling
- [WAGMI hooks](https://github.com/tmm/wagmi) for easy connection to Metamask

### CODE REVIEW

File: solutions/mint-nft/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# mint-nft example
+
+This example shows how -------------------------
```
Comment (lfades) [inline solutions/mint-nft/README.md:3]: This isn't filled

Comment (lfades) [inline solutions/mint-nft/README.md:3]: This isn't filled

File: solutions/mint-nft/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Demo
+
+https://mint-nft.vercel.app
```
Comment (lfades) [inline solutions/mint-nft/README.md:7]: This demo link doesn't seem to be the one for this example

File: solutions/mint-nft/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "vercel-example--mint-nft",
```
Comment (lfades) [inline solutions/mint-nft/package.json:2]:
```suggestion
  "name": "mint-nft",
```

File: solutions/mint-nft/components/UploadNft.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            role="alert"
+            className="flex w-full justify-center mt-4 text-transparent bg-clip-text bg-gradient-to-br from-pink-400 to-red-600"
+          >
+            {imageWarning}
+          </aside>
+        )}
+        <div className="mt-6 sm:items-center flex justify-center">
+          <Button disabled={disabled} size="lg" variant="black">
+            {loading ? <LoadingDots /> : 'Confirm'}
+          </Button>
```
Comment (lfades) [inline solutions/mint-nft/components/UploadNft.tsx:102]: What does this button do?

File: solutions/mint-nft/lib/fetcher.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    } else {
+      console.error('Network Error, CORS or timeout.')
+    }
+  }
+}
+
+export const getKey = (pageIndex: number = 1, previousPageData: any) => {
+  if (previousPageData && !previousPageData.length) return null // reached the end
+  return `api/nft?page=${pageIndex}` // SWR key
+}
```
Comment (lfades) [inline solutions/mint-nft/lib/fetcher.ts:32]: This file doesn't seem to be used

**Review by lfades — COMMENTED** [review]
Added some comments

File: solutions/mint-nft/README.md (hunk 2 above)
Comment (okbel) [reply solutions/mint-nft/README.md:7]: @gbibeaul

Comment (gbibeaul) [reply solutions/mint-nft/README.md:7]: https://github.com/vercel/examples/pull/152/files#diff-c19ce76276cdaf4483d6038fa8143b1dff3b9a1390b3ec30e7ce0b1b266c7dbfL3

**Comment by gbibeaul** [comment]
Resolved everything as I followed up here https://github.com/vercel/examples/pull/152

**Pull request merged**